### DISCUSSION TITLE
IPv4 Networking and Debian LXC

### DISCUSSION THREAD

**Discussion by SeanNazareth** [tteck/Proxmox#1358] [Help Needed]
Hello,

I came across your scripts for Proxmox and think they're very useful.  It has helped me to understand how to setup and use proxmox in many scenarios.

However, I've found that if I use your scripts to setup a Debian LXC, disable IPv6 (per the install script), and set IPv4 to DHCP, I keep losing my IPv4 IP address.  In my case, I am using VLANs, though I've tried disabling them and see the same issue.  The way I was setting things up was to update the /etc/network/interfaces file in the proxmox configuraiton as follows:

       auto vmbr1.10
       iface vmbr1.10 inet dhcp

        auto vmbr1.107
        iface vmbr1.107 inet dhcp

       auto vmbr1
       iface vmbr1 inet static
        address 0.0.0.0
        bridge-ports enp2s0
        bridge-stp off
        bridge-fd 0
        bridge-vlan-aware yes
        bridge-pvid 0
        bridge-vids 10,107

Furthermore, before I spin up the container, I add the two VLANs (10, 107) to the container as separate ethernet interfaces in the Proxmox UI under the network hardware option.  When setting up the container network option, the IPv6 is set to "static", but by not specifying an IP address, it seems to indicate that the IPv6 is disabled.  IPv6 was explicitly disabled in your setup script.

The container will get an initial IP address on both interfaces, but it will lose the address within 30-60 minutes or so.  While the IP address exists, network behavior of any docker containers, or even SSH work normally.  However, once it loses the IP address, I have to reboot the container to get another IP address.   If I set up a Ubuntu LXC container with 22.04, I don't seem to have this problem.  When I use your docker script, it uses the Debian LXC as a base, so currently, my docker (and portainer) stop working as a result.  I haven't yet tried the docker-alpine option that you provide.

Do you have any idea what I can do to get the DCHP for IPv4 in the Debian LXC container to be stable?

Thanks.

**Comment by emanuellopes** [comment]
Did you solve your issue?
